### ISSUE TITLE
Ambiguous Selectors

### ISSUE THREAD

**Issue by jehoshua02** [deseretdigital-ui/CSS-Guidelines#13]
From [CSS Selectors: Naming](https://github.com/deseretdigital-ui/CSS-Guidelines/blob/gh-pages/index.md#naming)

> My advice with regard to naming things in CSS is to pick a name that is sensible, but somewhat ambiguous: aim for high reusability.

Not sure I agree with "ambiguous" naming. Often our styles are not reusable without some heavy polishing.

I'd prefer specific class names with messy styles than ambiguous class names with messy styles. No sense naming it for reusability when it's not going to be worthy of reuse. This could make things more confusing because we see a class name that looks reusable but only find out it applies styles that don't make sense.

Then again, maybe if we were to use reusable names, we would be more careful to make sure the styles are reusable, paying very careful attention to selector intent and being very "Reusability-Minded".

I just feel like we should use more specific class names and reuse nothing by accident. And when we identify something we like to reuse, we pull it out, give it a good name, polish it up. That said, I agree with "Strive for reusable, recyclable classes".

Before Closing Issue:
- [x] Response from @jstoutenburg
- [x] Response from @janiv
- [ ] Response from @kbratko
- [x] Response from @jspencerDDM
- [ ] Consensus or Decision
- [ ] Pull request or other action if necessary

**Comment by rsahlstrom** [comment]
We really need to be focusing on reusable classes and abstracting as much as possible. Travis and Jim want a stylesheet where we can define components and use these components across the site. Reusable classes is an important part of this.

See https://smacss.com/

**Comment by rsahlstrom** [comment]
@deseretdigital-ui/front-end-developers

**Comment by jehoshua02** [comment]
I totally agree.

But instead of having one poorly named un-reusable class name, we could end up with dozens or poorly named and undocumented "reusable" classes that I would only be afraid to reuse or modify for fear of breaking things.

But we are just going to have to push through it. We will be spending a lot more time on Code Reviews.

**Comment by rsahlstrom** [comment]
I agree that the above situation would also be bad. We want neither.

**Comment by knitcodemonkey** [comment]
I don't think we should have .nav as a class name as it could mean every nav on the page. .homes-main-top-nav also seems like overkill. Perhaps the happy medium falls in between on .top-nav or .main-nav. I love the idea of having reusable css across products without having to modify class names to be less product specific.

**Comment by hhatfield** [comment]
I think this really speaks to why components are the right train of thought. I think it would be ok to have a `.primary-nav`, but I think `nav-dropdown` is more descriptive about what it does. Paired with #14, I think we should be able to strike the right balance.

**Comment by jehoshua02** [comment]
I'm not sold yet that making every little thing reusable is going to make the conceptual overhead, or the speed of development, or ease of maintenance any easier to deal with. Right now, we are nowhere near the level of reusability that we should be, so I'm willing to go along with it. Trying new things and gaining experience is valuable to me.
